scipy/special/_cdflib.pxd

```diff
@@ -31,6 +31,14 @@ cdef (double, double, int, double) cdff_which1(double f, double dfn, double dfd)
 cdef (double, int, double) cdff_which2(double p, double q, double dfn, double dfd) noexcept nogil
 cdef (double, int, double) cdff_which3(double p, double q, double f, double dfd) noexcept nogil
 cdef (double, int, double) cdff_which4(double p, double q, double f, double dfn) noexcept nogil
+cdef (double, double, int, double) cdffnc_which1(double f, double dfn, double dfd, double phonc) nogil
```

Comment:
Here you need `noexcept`s too. Don't know why but seriously slows down the execution since exceptions are potential Python interactions.